### PR TITLE
Buy album

### PR DESCRIPTION
**Pull request by gon-escalante**
## Summary

added album selling endpoint, and changed some of the users test to make part of it reusable

## Trello Card

https://trello.com/c/Cf4bwtvS/30-comprar-album

### CODE REVIEW

File: app/controllers/album.js [hunk 1]
```diff
@@ -17,3 +18,21 @@ exports.getAlbumPhotos = (req, res) =>
...
       return apiError(err.message);
     });
+
+exports.buyAlbum = (req, res, next) =>
+  getAlbum(req.params.id)
+    .then(album => {
+      if (!album) {
+        return next(notFoundError('Album not found'));
+      }
+      return getUserAlbums(req.user.id, next).then(albums => {
```
Comment (igna92ts) [inline app/controllers/album.js]: why do you send next as a parameter here?

File: app/services/user.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 const { user: User } = require('../models'),
   { databaseError } = require('../errors'),
-  { DEFAULT_PAGE_LIMIT } = require('../constants');
+  { DEFAULT_PAGE_LIMIT } = require('../constants'),
+  { userAlbum: UserAlbum } = require('../models');
 
 exports.createUser = ({ email, ...otherUserData }, next) =>
```
Comment (igna92ts) [inline app/services/user.js]: theres no need to call next in the services, you can just throw and the catch in the controller can get it and call next then, without the service needing to know what next is.

**Review by igna92ts — CHANGES_REQUESTED** [review]

**Review by igna92ts — APPROVED** [review]

File: test/controllers/album.test.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          done(err);
+        }
+        done();
+      }));
+  rpMock.mockRejectedValue(notFoundResponse);
+  it('should return an error if the album doesnt exist', done =>
+    request(app)
+      .post('/albums/999')
+      .query({ token: authorizedUserToken })
+      .expect(404)
```
Comment (igna92ts) [inline test/controllers/album.test.js:62]: how are you checking that errors are the ones you were expecting? this error could says `cannot read property importanteValidarEsto form undefined`

**Review by igna92ts — CHANGES_REQUESTED** [review]

File: test/controllers/album.test.js (hunk 3 above)
Comment (gon-escalante) [reply test/controllers/album.test.js:62]: 🍆

**Pull request closed without merging**